vulnerability-classes/injection/SSTI-Razor.bcheck

```diff
@@ -0,0 +1,27 @@
+metadata:
+    language: v2-beta
+    name: ".NET Razor SSTI Injection"
+    description: "Inserts a .NET Razor SSTI payload into each parameter to detect suspicious input evaluation"
+    author: "alp1n3.eth"
+    tags: "active", "injection", "template", "template injection", "scan", "ssti", "razor"
+# reference 1: https://clement.notin.org/blog/2020/04/15/Server-Side-Template-Injection-(SSTI)-in-ASP
+# reference 2: https://www.schtech.co.uk/razor-pages-ssti-rce/
+
+
+define:
+    calculation = "@(1337*1337)"
+    answer = "1787569"
+
+
+given query or body insertion point then
+    send payload:
```

Comment:
you could check for the answer in the base response before issuing the payload to mitigate against false positives.

Reply:
Fixed. Thank you for the tip! Changed to be: 
``` 
if not({answer} in {base.response}) then
	    send payload:
	        replacing: {calculation}
```